### PR TITLE
Disable suffix sharing for block tree index

### LINKED ISSUE #12702: Optimize FST suffix sharing for block tree index
### Description

Today our block tree index will encode `floordata` as output. The floor data is guaranteed to be stored within single arc (never be prefix shared) in FST because fp is encoded before it. As a suffix, floor data can rarely be shared too. I wonder if we can avoid adding floor data outputs into `NodeHash` some way?

Out of curiosity, i tried to completely disable suffix sharing in block tree index, result in only 1.47% total .tip size increased for `wikimediumall`.

<byte-sheet-html-origin data-id="1697792179685" data-version="4" data-is-embed="false" data-grid-line-hidden="false" data-importRangeRawData-spreadSource="https://bytedance.feishu.cn/sheets/DdtqsMJQjhLVCWtD2P4cvpyXnVe" data-importRangeRawData-range="&#39;Sheet1&#39;!F1:H2">

baseline | candidate |  diff
-- | -- | --
27188690 | 27588186 | 1.47%

### PR DESCRIPTION
**Pull request by gf2121**
closes https://github.com/apache/lucene/issues/12702

### CODE REVIEW

File: lucene/CHANGES.txt [hunk 1]
```diff
@@ -227,6 +227,8 @@ Optimizations
 
 * GITHUB#12712: Speed up sorting postings file with an offline radix sorter in BPIndexReader. (Guo Feng)
 
+* GITHUB#12702: Disable suffix sharing for block tree index. (Guo Feng)
```
Comment (mikemccand) [inline lucene/CHANGES.txt:230]: Maybe also add something more user friendly, e.g. "making writing the terms dictionary index faster and less RAM hungry, while making the index a bit (~1.X
% for the terms index file on wikipedia) bigger"?

**Review by mikemccand — APPROVED** [review]
Thanks @gf2121!

File: lucene/CHANGES.txt (hunk 1 above)
Comment (gf2121) [reply lucene/CHANGES.txt]: Great description! I just copied it :)

**Comment by mikemccand** [comment]
Unfortunately, because I messed up conflict resolution on backport of #12633 (commit https://github.com/apache/lucene/commit/2ca906d99c64fc45dd4246b7daad57db3a7abdf8), I accidentally reverted this change!  This means it did not in fact make it into 9.9 :(

I uncovered this (and will correct it) while backporting #12624 ...

**Pull request merged**